### PR TITLE
Refactor s2s bridge

### PR DESCRIPTION
**Pull request by fewensa**
Refactor s2s bridge.

This pr is an exploration of s2s bridge implementations. also only includes the delivery of messages.
Just include bridge pangolin-pangoro, other s2s bridges and other features (such as feemarket/docs) will be improved in the future.

### CODE REVIEW

**Comment by HackFisher** [comment]
How to update the Cargo.lock in bridges folder in one command?

**Comment by HackFisher** [comment]
Which s2s bridges are refactored to use the new approach?

All of them or some of them?

File: bridges/pangolin-pangoro/bridge/src/command/handler/init.rs [hunk 1]
```diff
@@ -1,129 +1,56 @@
...
-            }
+async fn init_bridge(
+    bridge: BridgeName,
+    config_pangolin: ChainInfoConfig,
+    config_pangoro: ChainInfoConfig,
+) -> color_eyre::Result<()> {
+    let client_pangolin = PangolinClientComponent::component(config_pangolin.into()).await?;
+    let client_pangoro = PangoroClientComponent::component(config_pangoro.into()).await?;
+    let hash = match bridge {
+        BridgeName::PangoroToPangolin => {
```
Comment (HackFisher) [inline bridges/pangolin-pangoro/bridge/src/command/handler/init.rs:27]: Wrong direction?

File: frame/assistants/client-pangoro/src/fastapi/s2s/mod.rs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pub mod generic;
+#[cfg(feature = "s2s-pangolin")]
+pub mod pangolin;
```
Comment (HackFisher) [inline frame/assistants/client-pangoro/src/fastapi/s2s/mod.rs:3]: Same

File: frame/assistants/client-pangoro/src/fastapi/s2s/pangolin.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+use crate::client::PangoroClient;
+
+impl PangoroClient {}
```
Comment (HackFisher) [inline frame/assistants/client-pangoro/src/fastapi/s2s/pangolin.rs:3]: Type name and file name mismatch.

File: frame/assistants/client-pangolin/src/fastapi/s2s/mod.rs [hunk 4]
```diff
@@ -0,0 +1,3 @@
+pub mod generic;
+#[cfg(feature = "s2s-pangoro")]
+pub mod pangoro;
```
[the remark this answers is not in the document]
Comment (fewensa) [reply frame/assistants/client-pangolin/src/fastapi/s2s/mod.rs:3]: I'm not very sure about this. The original idea was to wrap the s2s related methods. But it was used directly in the end. So. This empty file is not actually used.
https://github.com/darwinia-network/bridger/blob/f01fa63a534d131abcec62387c112b59e350b430/bridges/pangolin-pangoro/bridge/src/service/message/pangolin_to_pangoro/delivery_relay.rs#L219-L225

[the remark this answers is not in the document]
Comment (fewensa) [reply frame/assistants/client-pangolin/src/fastapi/s2s/mod.rs:3]: I think remove it currently is better, add it if required later

**Pull request merged**